### PR TITLE
Create better plans for NULL comparisons

### PR DESCRIPTION
**Pull request by systay**

### CODE REVIEW

File: go/vt/vtgate/engine/route_test.go [hunk 1]
```diff
@@ -148,6 +150,40 @@ func TestSelectEqualUnique(t *testing.T) {
...
+	expectResult(t, "sel.Execute", result, &sqltypes.Result{})
+
+	vc.Rewind()
+
+	result, err = sel.Execute(vc, map[string]*querypb.BindVariable{}, true)
+	require.NoError(t, err)
+	vc.ExpectLog(t, []string{
+		`ResolveDestinations ks [] Destinations:DestinationAnyShard()`,
+		`ExecuteMultiShard ks.-20: dummy_select_field {} false false`,
+	})
```
Comment (harshit-gangal) [inline go/vt/vtgate/engine/route_test.go:183]: why this is expected when it is not sent to a tablet?

File: go/vt/vtgate/planbuilder/route_option.go [hunk 2]
```diff
@@ -278,6 +284,25 @@ func (ro *routeOption) computeEqualPlan(pb *primitiveBuilder, comparison *sqlpar
...
+// computeEqualPlan computes the plan for an equality constraint.
+func (ro *routeOption) computeISPlan(pb *primitiveBuilder, comparison *sqlparser.IsExpr) (opcode engine.RouteOpcode, vindex vindexes.SingleColumn, condition sqlparser.Expr) {
+	// we only handle IS NULL correct. IsExpr can contain other expressions as well
+	if comparison.Operator != sqlparser.IsNullStr {
+		return engine.SelectScatter, nil, nil
+	}
+
+	left := comparison.Expr
+
+	vindex = ro.FindVindex(pb, left)
```
Comment (harshit-gangal) [inline go/vt/vtgate/planbuilder/route_option.go:296]: nit: this can be passed in the method call

File: go/vt/vtgate/engine/route_test.go (hunk 1 above)
Comment (harshit-gangal) [reply go/vt/vtgate/engine/route_test.go:183]: Ignore this comment. I see two tests with wantfields false and true.

**Comment by deepthi** [comment]
nit: it will be nice to document the problem in the PR description.

**Pull request merged**